### PR TITLE
Use the heartbeats shipped with the digest for cluster membership and failure detection

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: chitchat/src/failure_detector.rs [hunk 1]
```diff
@@ -55,23 +41,30 @@ impl FailureDetector {
...
+            .phi(chitchat_id)
+            .map(|phi| phi <= self.config.phi_threshold)
+            .unwrap_or(false);
+        debug!(node_id=%chitchat_id.node_id, phi=?phi_opt, is_alive=is_alive, "computing node liveness");
+        if is_alive {
+            self.live_nodes.insert(chitchat_id.clone());
+            self.dead_nodes.remove(chitchat_id);
+        } else {
+            self.live_nodes.remove(chitchat_id);
+            if !self.dead_nodes.contains_key(&chitchat_id) {
```
Comment (fulmicoton) [inline chitchat/src/failure_detector.rs:61]: this was a bug.
When a node was identified twice as dead, we would update its time of death.

File: chitchat/src/state.rs [hunk 2]
```diff
@@ -303,22 +317,40 @@ impl ClusterState {
 
     pub(crate) fn apply_delta(&mut self, delta: Delta) {
         // Remove nodes to reset.
+        if delta.nodes_to_reset.len() > 0 {
+            tracing::error!(nodes_to_reset=?delta.nodes_to_reset, "nodes to reset");
```
Comment (fmassot) [inline chitchat/src/state.rs:321]: Is having to reset nodes an error? info or warning would be better

File: chitchat/src/state.rs [hunk 3]
```diff
@@ -303,22 +317,40 @@ impl ClusterState {
...
                 key_values,
             } = node_delta;
-            let node_state = self
-                .node_states
-                .entry(chitchat_id.clone())
-                .or_insert_with(|| NodeState::new(chitchat_id, self.listeners.clone()));
-            if node_state.heartbeat < heartbeat {
-                node_state.heartbeat = heartbeat;
+            let node_state = self.node_state_mut(&chitchat_id);
+            if node_state.last_gc_version == 0u64 {
```
Comment (fmassot) [inline chitchat/src/state.rs:334]: Very helpful explanation.

File: chitchat/src/state.rs (hunk 2 above)
Comment (fulmicoton) [reply chitchat/src/state.rs]: yes that was a mistake. I made it an info.

**Pull request merged**